frontend/src/metabase/dashboard/components/PublicLinkPopover/DashboardPublicLinkPopover.unit.spec.tsx

```diff
@@ -0,0 +1,99 @@
+import { useState } from "react";
+import fetchMock from "fetch-mock";
+import userEvent from "@testing-library/user-event";
+import { renderWithProviders, screen } from "__support__/ui";
+import {
+  createMockSettingsState,
+  createMockState,
+} from "metabase-types/store/mocks";
+import { createMockDashboard, createMockUser } from "metabase-types/api/mocks";
+import { setupDashboardPublicLinkEndpoints } from "__support__/server-mocks";
+import { DashboardPublicLinkPopover } from "metabase/dashboard/components/PublicLinkPopover/DashboardPublicLinkPopover";
+import type { Dashboard } from "metabase-types/api";
+
+const SITE_URL = "http://metabase.test";
+const TEST_DASHBOARD_ID = 1;
+
+const TestComponent = ({
+  dashboard,
+  onClose: onCloseMock,
+}: {
+  dashboard: Dashboard;
+  onClose: () => void;
+}) => {
+  const [isOpen, setIsOpen] = useState(true);
+
+  const onClose = () => {
+    setIsOpen(false);
+    onCloseMock();
+  };
+
+  return (
+    <DashboardPublicLinkPopover
+      dashboard={dashboard}
+      isOpen={isOpen}
+      onClose={onClose}
+      target={<button>Target</button>}
+    />
+  );
+};
+
+const setup = ({
+  hasPublicLink = true,
+}: {
+  hasPublicLink?: boolean;
+} = {}) => {
+  const TEST_DASHBOARD = createMockDashboard({
+    id: TEST_DASHBOARD_ID,
+    public_uuid: hasPublicLink ? "mock-uuid" : null,
+  });
+
+  setupDashboardPublicLinkEndpoints(TEST_DASHBOARD_ID);
+
+  const state = createMockState({
+    currentUser: createMockUser({ is_superuser: true }),
+    settings: createMockSettingsState({
+      "site-url": SITE_URL,
+    }),
+  });
+
+  const onClose = jest.fn();
+
+  renderWithProviders(
+    <TestComponent dashboard={TEST_DASHBOARD} onClose={onClose} />,
+    {
+      storeInitialState: state,
+    },
+  );
+};
+describe("DashboardPublicLinkPopover", () => {
+  it("should display a question-specific public url", async () => {
+    setup();
+
+    expect(
+      await screen.findByText(`${SITE_URL}/public/dashboard/mock-uuid`),
+    ).toBeInTheDocument();
+  });
+
+  it("should not display extensions for the public link", () => {
+    setup();
+
+    expect(screen.queryByTestId("extension-option")).not.toBeInTheDocument();
+  });
+
+  it("should call Dashboard public link API when creating link", () => {
+    setup({ hasPublicLink: false });
+
+    expect(
+      fetchMock.calls(`path:/api/dashboard/${TEST_DASHBOARD_ID}/public_link`),
```

Comment:
```suggestion
      fetchMock.calls(`path:/api/dashboard/${TEST_DASHBOARD_ID}/public_link`, {
        method: "POST",
      }),
```

I think we should make assertions more explicit. Now we don't assert what method this API is called with. It could be confused whether we're trying to get a public_link or actually setting this dashboard to be publically available.

Reply:
Ah ok I wanted to do something like this but I wasn't even sure if this was something we could do. Thanks for that